generators/typescript/codegen/src/ast/TypeLiteral.ts

```diff
@@ -0,0 +1,169 @@
+import { assertNever } from "@fern-api/core-utils";
+import { AstNode, Writer } from "./core";
+import { Type } from "./Type";
+
+type InternalTypeLiteral = Array_ | Boolean_ | Number_ | Object_ | String_ | Tuple;
+
+interface Array_ {
+    type: "array";
+    valueType: Type;
+    values: TypeLiteral[];
+}
+
+interface Boolean_ {
+    type: "boolean";
+    value: boolean;
+}
+
+interface Number_ {
+    type: "number";
+    value: number;
+}
+
+interface Object_ {
+    type: "object";
+    fields: ObjectField[];
+}
+
+interface ObjectField {
+    name: string;
+    valueType: Type;
+    value: TypeLiteral;
+}
+
+interface String_ {
+    type: "string";
+    value: string;
+}
+
+interface Tuple {
+    type: "tuple";
+    // TODO: In theory this should be a tuple type, not an array of types
+    valueTypes: Type[];
+    values: TypeLiteral[];
+}
```

Comment:
I wonder if the `value` and `valueType` should be a single interface so that it's impossible for the length to be inconsistent.